### DISCUSSION TITLE
TCA package does not build on Xcode 13

### DISCUSSION THREAD

**Discussion by ferologics** [pointfreeco/swift-composable-architecture#828] [Q&A]
When I try building the `ComposableArchitecture` for `Any iOS Device` on new Xcode 13 I get a lot of errors. Is this me or is there some WIP for new Xcode compatibility? This is on latest `main` (`fa4fa7667bc97bb31bfa0ced10327ccb3c4cfcc2`). This is on my Intel based machine.

<img width="1888" alt="image" src="https://user-images.githubusercontent.com/5576161/134654885-23d72369-e179-46d4-9cc3-c3731369653c.png">

**Comment by ferologics** [comment]
This doesn't happen on my M1 MacBook, so perhaps it's just Xcode 13 tripping a bit...

**Comment by stephencelis** [comment]
Have you tried updating your SPM packages? A few seem to be a little out of date. For example, combine-schedulers is at 0.5.3.

**Reply by ferologics** [reply]
Good point, I haven't used the "update packages to latest versions" Xcode SPM button... AFAIK it's up to Xcode to resolve the right package version based on TCA package manifest (starting `from: xyz`) -- so maybe it's just a mishap on that front with the cache acting up. Ideally, TCA would (or is already?) be pinned to the latest versions of the packages it depends on so there's no room for error here. Anyways, I [switched to my M1 mac](https://github.com/pointfreeco/swift-composable-architecture/discussions/828#discussioncomment-1379948) and everything compiled just fine 👍

**Comment by BSEBAT1** [comment]
I have the exact same situation. I was finally able to get the app to build by deleting derived data then reset swift package cache. However I now cannot archive the application. 

Of course this does not even begin the count the number of bugs I have had on iOS 15. I am constantly gettin crashes from the composable framework with no logs and failures in reducers.

**Reply by stephencelis** [reply]
@BSEBAT1 That's concerning to hear! Do you have any stack traces that point to specific parts of the framework?

One common issue that folks may encounter in apps with unified app state (like TCA, ReSwift, and others) is having state that becomes too large:

- https://github.com/pointfreeco/swift-composable-architecture/discussions/488
- https://github.com/pointfreeco/swift-composable-architecture/discussions/752

If this is the cause of the issue, you may need to box some of your app's state in reference types, as suggested in the above threads.

**Comment by erikolsson** [comment]
While I can't offer a clear solution, I ran into the same build issue a few days ago.
I think it's an issue related to SPM caches.

Apart from clearing derived data I also force updated all dependencies to latest (TCA imports `swift-identified-collections` v0.1.0 by default I think) and eventually it just worked :)

**Reply by K3TZR** [reply]
When I updated to Xcode 13 I had a similar problem in an app I'm building. Xcode 13 had errors related to AttributedString and UNNotificationInterruptionLevel.

I did a "Reset Package Caches" (which crashed Xcode 13), restarted Xcode 13 and did a "Update to Latest Package Versions" and now the errors are gone.
